nrf-hal-common/src/wdt.rs

```diff
@@ -0,0 +1,314 @@
+                Some(WatchdogHandle(Hdl4))
+            } else {
+                None
+            },
+            hdl5: if handles >= NumHandles::Six {
+                Some(WatchdogHandle(Hdl5))
+            } else {
+                None
+            },
+            hdl6: if handles >= NumHandles::Seven {
+                Some(WatchdogHandle(Hdl6))
+            } else {
+                None
+            },
+            hdl7: if handles == NumHandles::Eight {
+                Some(WatchdogHandle(Hdl7))
+            } else {
+                None
+            },
+        }
+    }
+
+    /// Enable the watchdog interrupt
+    ///
+    /// NOTE: Although the interrupt will occur, there is no way to prevent
+    /// the reset from occuring. From the time the event was fired, the
+    /// system will reset two LFCLK ticks later (61 microseconds) if the
+    /// interrupt has been enabled.
+    #[inline(always)]
+    pub fn enable_interrupt(&mut self) {
+        self.wdt.intenset.write(|w| w.timeout().set_bit());
+    }
+
+    /// Disable the watchdog interrupt
+    ///
+    /// NOTE: This has no effect on the reset caused by the Watchdog
+    #[inline(always)]
+    pub fn disable_interrupt(&mut self) {
+        self.wdt.intenclr.write(|w| w.timeout().set_bit());
+    }
+
+    /// Set the number of 32.768kHz ticks in each watchdog period
+    ///
+    /// This value defaults to 0xFFFF_FFFF (1.5 days) on reset.
+    ///
+    /// Note: there is a minimum of 15 ticks (458 microseconds). If a lower
+    /// number is provided, 15 ticks will be used as the configured value
+    #[inline(always)]
+    pub fn set_lfosc_ticks(&mut self, ticks: u32) {
+        self.wdt
+            .crv
+            .write(|w| unsafe { w.bits(ticks.max(0x0000_000F)) });
+    }
+
+    /// Should the watchdog continue to count during sleep modes?
+    ///
+    /// This value defaults to ENABLED on reset.
+    pub fn run_during_sleep(&self, setting: bool) {
+        self.wdt.config.modify(|_r, w| w.sleep().bit(setting));
+    }
+
+    /// Should the watchdog continue to count when the CPU is halted for debug?
+    ///
+    /// This value defaults to DISABLED on reset.
+    pub fn run_during_debug_halt(&self, setting: bool) {
+        self.wdt.config.modify(|_r, w| w.halt().bit(setting));
+    }
+}
+
+impl Watchdog<Active> {
+    /// Is the watchdog still awaiting pets from any handle?
+    ///
+    /// This reports whether sufficient pets have been received from all
+    /// handles to prevent a reset this time period
+    #[inline(always)]
+    pub fn awaiting_pets(&self) -> bool {
+        let enabled = self.wdt.rren.read().bits();
+        let status = self.wdt.reqstatus.read().bits();
+        (status & enabled) == 0
+    }
+}
+
+impl<T> Watchdog<T>
+where
+    T: sealed::WdMode,
+{
+    /// Is the watchdog active?
+    #[inline(always)]
+    pub fn is_running(&self) -> bool {
+        // TODO: Should we just believe the type state?
```

Comment:
Since the watchdog can apparently be running after reset, it seems like the typestate is actually incorrect, so why not remove it?

Reply:
The typestate is mainly there to prevent trying to configure the device after activation (which will have no effect). The watchdog will only be running after a start if there was some kind of soft reset (I'm not sure what kind of resets will or won't clear the watchdog state).

I thought about checking if the watchdog is already active in `Watchdog::new()`, though I'm not sure what the right response there is, other than to panic, or just block until it expires, but that could be bad if the watchdog is set to something huge, like 1.5 days.